### PR TITLE
worker: delete taskState when task is stopped/finished

### LINKED ISSUE #1594: task state is not deleted in prometheus when task is stopped
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
When task on dm-worker is re-scheduled, the task state on the original worker will become `stopped` unless this worker is restarted, which looks confusing.

2. What did you expect to see?
This prometheus state is deleted.

3. What did you see instead?
It becomes `stopeed`.

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.2
        ```

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
fix #1594

### What is changed and how it works?
delete taskState when task is stopped/finished

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

When task is running:
```
dm_worker_cpu_usage 3.528409857613362
# HELP dm_worker_task_state state of task, 0 - invalidStage, 1 - New, 2 - Running, 3 - Paused, 4 - Stopped, 5 - Finished
# TYPE dm_worker_task_state gauge
dm_worker_task_state{source_id="mysql-replica-01",task="testdm"} 2
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
```
After task is stopped:
```
# HELP dm_worker_cpu_usage the cpu usage of worker
# TYPE dm_worker_cpu_usage gauge
dm_worker_cpu_usage 3.521312888533776
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 2.6124e-05
go_gc_duration_seconds{quantile="0.25"} 3.0232e-05
```

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
there's a check_metric script, which could be used in tests.

File: dm/worker/subtask.go [hunk 1]
```diff
@@ -692,3 +693,11 @@ func (st *SubTask) HandleError(ctx context.Context, req *pb.HandleWorkerErrorReq
 	}
 	return err
 }
+
+func updateTaskState(task, sourceID string, stage pb.Stage) {
+	if stage == pb.Stage_Stopped || stage == pb.Stage_Finished {
+		taskState.DeleteAllAboutLabels(prometheus.Labels{"task": task, "source_id": sourceID})
```
Comment (lance6716) [inline dm/worker/subtask.go:699]: we might update the description of metics in dm.json and dm_instance.json

"The current state of subtasks..."

File: tests/_utils/check_metric_not_contains [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    metric=$(curl -s http://127.0.0.1:$port/metrics | grep $metric_name | wc -l)
+    if [ $metric -eq 0 ]; then
+        exit 0
+    fi
+    ((counter+=1))
+    echo "wait for valid metric for $counter-th time"
+    sleep 1
+done
+
+echo "metric $metric_name has invalid value $metric"
```
Comment (lance6716) [inline tests/_utils/check_metric_not_contains:25]:
```suggestion
echo "metric $metric_name has invalid count $metric"
```

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — COMMENTED** [review]
Please resolve conflict. rest LGTM

**Comment by lance6716** [comment]
```
[2021-04-20T07:01:21.280Z] tests/_utils/check_metric: line 20: [: 1: unary operator expected

[2021-04-20T07:01:21.280Z] wait for valid metric for 1-th time
```

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Comment by lichunzhu** [comment]
integration test shardddl1 seems still unstable:
https://internal.pingcap.net/idc-jenkins/job/dm_ghpr_test/8962/display/redirect

**Pull request merged**